### PR TITLE
[SHOP] Refactor the security template and split it into twig hooks

### PR DESCRIPTION
**Pull request by TheMilek**
| Q               | A
|-----------------|-----
| Branch?         | bootstrap-shop 
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ShopBundle/Resources/config/app/twig_hooks/base.yaml [hunk 1]
```diff
@@ -29,12 +29,58 @@ twig_hooks:
...
+            desktop:
+                template: '@SyliusShop/shared/layout/base/header/content/security/logged_in_user/desktop.html.twig'
+                
+        'sylius_shop.base.header.content.security.logged_in_user.mobile':
+            dropdown_button:
+                template: '@SyliusShop/shared/layout/base/header/content/security/logged_in_user/mobile/dropdown_button.html.twig'
+            dropdown_menu:
+                template: '@SyliusShop/shared/layout/base/header/content/security/logged_in_user/mobile/dropdown_menu.html.twig'
+
+        'sylius_shop.base.header.content.security.logged_in_user.mobile.dropdown_menu':
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ShopBundle/Resources/config/app/twig_hooks/base.yaml]:
```suggestion
        'sylius_shop.base.header.content.security.logged_in_user.mobile.menu':
```

File: src/Sylius/Bundle/ShopBundle/Resources/config/app/twig_hooks/base.yaml [hunk 2]
```diff
@@ -29,12 +29,58 @@ twig_hooks:
...
+            dashboard_button:
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ShopBundle/Resources/config/app/twig_hooks/base.yaml]:
```suggestion
            my_account:
```

File: src/Sylius/Bundle/ShopBundle/Resources/config/app/twig_hooks/base.yaml [hunk 3]
```diff
@@ -29,12 +29,58 @@ twig_hooks:
...
+                
+        'sylius_shop.base.header.content.security.logged_in_user':
+            mobile:
+                template: '@SyliusShop/shared/layout/base/header/content/security/logged_in_user/mobile.html.twig'
+            desktop:
+                template: '@SyliusShop/shared/layout/base/header/content/security/logged_in_user/desktop.html.twig'
+                
+        'sylius_shop.base.header.content.security.logged_in_user.mobile':
+            dropdown_button:
+                template: '@SyliusShop/shared/layout/base/header/content/security/logged_in_user/mobile/dropdown_button.html.twig'
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ShopBundle/Resources/config/app/twig_hooks/base.yaml]: IMO we can put it in the dropdown template below, what's the point of having it separate?

File: src/Sylius/Bundle/ShopBundle/Resources/config/app/twig_hooks/base.yaml [hunk 4]
```diff
@@ -29,12 +29,58 @@ twig_hooks:
...
+            desktop:
+                template: '@SyliusShop/shared/layout/base/header/content/security/visitor/desktop.html.twig'
+        
+        'sylius_shop.base.header.content.security.visitor.desktop':
+            user_icon:
+                template: '@SyliusShop/shared/layout/base/header/content/security/visitor/desktop/user_icon.html.twig'
+            login_button:
+                template: '@SyliusShop/shared/layout/base/header/content/security/visitor/desktop/login_button.html.twig'
+            register_button:
+                template: '@SyliusShop/shared/layout/base/header/content/security/visitor/desktop/register_button.html.twig'
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ShopBundle/Resources/config/app/twig_hooks/base.yaml]:
```suggestion
            login:
                template: '@SyliusShop/shared/layout/base/header/content/security/visitor/desktop/login_button.html.twig'
            register:
                template: '@SyliusShop/shared/layout/base/header/content/security/visitor/desktop/register_button.html.twig'
```
What's with all the suffixes?

File: src/Sylius/Bundle/ShopBundle/templates/shared/layout/base/header/content/security/logged_in_user/mobile/dropdown_button.html.twig [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{% from '@SyliusShop/shared/icon.html.twig' import icon as icon %}
+{% import '@SyliusShop/shared/buttons.html.twig' as buttons %}
```
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/templates/shared/layout/base/header/content/security/logged_in_user/mobile/dropdown_button.html.twig]: Is it used?
```suggestion
```

File: src/Sylius/Bundle/ShopBundle/templates/shared/layout/base/header/content/security/logged_in_user/mobile.html.twig [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{% from '@SyliusShop/shared/icon.html.twig' import icon as icon %}
+{% import '@SyliusShop/shared/buttons.html.twig' as buttons %}
+
```
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/templates/shared/layout/base/header/content/security/logged_in_user/mobile.html.twig]:
```suggestion
```

**Review by GSadee — APPROVED** [review]

**Pull request merged**